cli/azd/pkg/project/service_target_containerapp.go

```diff
@@ -103,6 +106,13 @@ func (at *containerAppTarget) Publish(
 				return
 			}
 
+			// Get ACR Login Server
+			loginServer, err := at.containerHelper.LoginServer(ctx)
```

Comment:
Building upon the other comment, by introducing a `helper` we've hidden awway important `env` state in a helper object that gets injected into DI. This complicates the code quite a bit.